simulation/costs.py

```diff
@@ -331,3 +331,43 @@ def estimate_boiler_upgrade_scheme_grant(
 
     if heating_system == HeatingSystem.HEAT_PUMP_GROUND_SOURCE:
         return 7_500
+
+
+def estimate_extended_boiler_upgrade_scheme_grant(
+    heating_system: HeatingSystem,
+    model: "DomesticHeatingABM",
+):
+
+    if heating_system not in [
+        HeatingSystem.HEAT_PUMP_AIR_SOURCE,
+        HeatingSystem.HEAT_PUMP_GROUND_SOURCE,
+    ]:
+        return 0
+
+    model_population_scale = ENGLAND_WALES_HOUSEHOLD_COUNT_2020 / model.household_count
+    boiler_upgrade_funding_cap_gbp = 450_000_000 / model_population_scale
+    if (
+        model.boiler_upgrade_scheme_cumulative_spend_gbp
+        >= boiler_upgrade_funding_cap_gbp
+    ):
+        return 0
+
+    if (
+        not datetime.date(2022, 4, 1)
+        <= model.current_datetime.date()
+        < datetime.date(2035, 4, 1)
+    ):
+        return 0
+
+    if (
+        not datetime.date(2022, 4, 1)
```

Comment:
Is this the definition of 'taper':
- now until 2028: 7_500
- 2028 - 2035: 5000
- after 2035: 0?

Reply:
Yes.

Reply:
I have updated the PR description.